### DISCUSSION TITLE
Imported files

### DISCUSSION THREAD

**Discussion by baltov** [juicedata/juicefs#945] [Q&A] [answered]
Hello, I am interesting how imported files are handled?

**Comment by davies** [comment, marked as answer]
@baltov The community version of JuiceFS can't import objects, only the hosted one can. The `import` rebuild the metadata for objects in object store, then you can read them from the mount point. You can also change the names and permissions, but can't modify the content. When you delete the imported files, the objects will NOT be deleted from object store.

**Reply by baltov** [reply]
Thank you for your nice answer. Would you like to share more information, how the metadata is rebuilded? :) Are you using other storage over cached store or the data is not cached? Thank you

**Reply by davies** [reply]
The metadata of imported objects looks like normal files (having name and all the attributes), except the content of the file is not a list of chunks, but a name in object store. Since the objects could be modified outside JuiceFS, so we can't cached the data of imported files, so it will be slower than files written from JuiceFS.

**Reply by baltov** [reply]
Thank you
